lib/Entity/Layout.php

```diff
@@ -987,6 +988,86 @@ public function toXlf()
+//
+//                    ON	ON	    INHERIT	YES     Media takes precedence      // Match - 3
+//                    ON	OFF	    INHERIT	NO      Media takes precedence      // Match - 4
+//                    ON	INHERIT	INHERIT	YES     Media takes precedence and Inherited from Layout        // Match - 5
+//
+//                    OFF	ON	    INHERIT	YES     Media takes precedence      // Match - 3
+//                    OFF	OFF	    INHERIT	NO      Media takes precedence      // Match - 4
+//                    OFF	INHERIT	INHERIT	NO      Media takes precedence and Inherited from Layout        // Match - 6
+//
+//                Widget Without Media
+//                LAYOUT	WIDGET		Widget stats collected?
+//                    ON	ON		    YES	    Widget takes precedence     // Match - 1
+//                    ON	OFF		    NO	    Widget takes precedence     // Match - 2
+//                    ON	INHERIT		YES	    Inherited from Layout       // Match - 7
+//                    OFF	ON		    YES	    Widget takes precedence     // Match - 1
+//                    OFF	OFF		    NO	    Widget takes precedence     // Match - 2
+//                    OFF	INHERIT		NO	    Inherited from Layout       // Match - 8
+
+                // Layout stat collection flag
+                $layoutEnableStat = $this->enableStat;
+
+                // Widget stat collection flag
+                $widgetEnableStat = $widget->getOptionValue('enableStat', $this->config->getSetting('WIDGET_STATS_ENABLED_DEFAULT'));
+
+                $enableStat = 0; // Match - 0
+
+                if ($widgetEnableStat == 'On') {
+                    $enableStat = 1; // Match - 1
+                } else if ($widgetEnableStat == 'Off') {
+                    $enableStat = 0; // Match - 2
+                } else if ($widgetEnableStat == 'Inherit') {
+
+                    try {
+                        // Media enable stat flag - WIDGET WITH MEDIA
+                        $media = $this->mediaFactory->getById($widget->getPrimaryMediaId());
+                        $mediaEnableStat = $media->enableStat;
+
+                        if ($mediaEnableStat == 'On') {
+                            $enableStat = 1; // Match - 3
+                        } else if ($mediaEnableStat == 'Off') {
+                            $enableStat = 0; // Match - 4
+                        } else if ($mediaEnableStat == 'Inherit') {
+                            if ($layoutEnableStat == 1) {
+                                $enableStat = 1; // Match - 5
+                            } else {
+                                $enableStat = 0; // Match - 6
+                            }
+                        }
+                    } catch (\Exception $e) { //  - WIDGET WITHOUT MEDIA
+                        $this->getLog()->error($widget->widgetId. ' is not a library media and does not have a media id.');
+                        if ($layoutEnableStat == 1) {
```

Comment:
can we make this simpler? just use the `$layoutEnableStat`

Reply:
Sorry, I don't get what you are trying to say?

Reply:
so I think you can remove the last `if` statement - because you are basically setting the final value to the condition you are testing?

Reply:
Got it now. 
` $enableStat = $layoutEnableStat;  // Match - 7 and 8`

I have also changed the if statement in line 1052 for Match 5 and 6
          
>                    try {
>                         // Media enable stat flag - WIDGET WITH MEDIA
>                         $media = $this->mediaFactory->getById($widget->getPrimaryMediaId());
>                         $mediaEnableStat = $media->enableStat;
> 
>                         if ($mediaEnableStat == 'On') {
>                             $enableStat = 1; // Match - 3
>                         } else if ($mediaEnableStat == 'Off') {
>                             $enableStat = 0; // Match - 4
>                         } else if ($mediaEnableStat == 'Inherit') {
>                             $enableStat = $layoutEnableStat;  // Match - 5 and 6
>                         }
>                     } catch (\Exception $e) { //  - WIDGET WITHOUT MEDIA
>                         $this->getLog()->error($widget->widgetId. ' is not a library media and does not have a media id.');
>                         $enableStat = $layoutEnableStat;  // Match - 7 and 8
>                     }